### PR TITLE
[Feature] Change hive meta load pool to improve perfomance

### LINKED ISSUE #5836: FE get stuck when there are logs of hive table query
### Real behavior (Required)
Use database got stuck.
The hive cache loading thread pool has been full used.

### StarRocks version (Required)
 - 2.1.5

### LINKED ISSUE #5874: Fe HiveMetacache Executor thread pool
## Feature request
Change Fe HiveMetacache Executor thread pool type

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/5874
Fixes #5836

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveRepository.java|1|1|100.00%|

**Review by imay — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveRepository.java [hunk 1]
```diff
@@ -38,7 +38,7 @@ public class HiveRepository {
     Map<String, HiveMetaCache> metaCaches = Maps.newHashMap();
     ReadWriteLock metaCachesLock = new ReentrantReadWriteLock();
 
-    Executor executor = Executors.newFixedThreadPool(100);
+    Executor executor = Executors.newCachedThreadPool();
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveRepository.java:41]:
```suggestion
    Executor executor = return new ThreadPoolExecutor(<corePoolSize>, Integer.MAX_VALUE, 60L, TimeUnit.SECONDS, new SynchronousQueue<Runnable>());
```
<corePoolSize> need greater than 0. like 50 or 100

Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/external/hive/HiveRepository.java:41]: executor threads is sed by `Config.hive_meta_cache_load_threads`, default value is 50

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveRepository.java|1|1|100.00%|

**Review by stephen-shelby — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Pull request merged**